public/js/NeighbourPlot.js

```diff
@@ -59,103 +62,167 @@ function NeighbourPlot(sampleData, element) {
+        this.svg.selectAll('circle')
+            .classed('filtered', true);
 
-    // define tooltip function and append to SVG
-    if(!this.tip) {
-        this.tip = d3.tip()
-            .attr('class', 'd3-tip')
-            .offset([-10, 0])
-            .html(function(d) {
-                var treatment = d.treatment || d.gene_name;
-
-                return '<p>ID: ' + d._id + '</p>' +
-                    '<p>Gene: ' + treatment + '</p>';
-            });
-        this.svg.call(this.tip);
+        // until proven otherwise..
+        this.svg.selectAll(tags)
+            .classed('filtered', false)
+            .moveToFront();
     }
+};
 
-    // draw plot axis
-    this.drawAxis();
-
-    // draw plot labels
-    this.drawPlotLabels('Component 1', 'Component 2');
-
-    // draw scatterplot points
-    this.drawPoints();
+/**
+ * draw: Draw the lineplot.
+ *
+ * @this {NeighbourPlot}
+ */
+NeighbourPlot.prototype.draw = function() {
+    d3.select(this.element + ' > svg').remove(); // clear canvas
+    this._addBackground();
+    this._setScale();
+    this._drawAxis();
+    this._defineToolTip();
+    this._drawPoints();
+    this._drawAxisLabels();
+};
 
+/**
+ * update: Change the dimensionality reduction used to display plot points.
+ *
+ * @this {NeighbourPlot}
+ * @param {string} reductionType - The dimensionality reduction method
+ *     to be used to display the sample points. Should be one of 'pca'
+ *     or 'tsne'
+ */
+NeighbourPlot.prototype.update = function(reductionType) {
+    this.reductionType = reductionType;
+    this._setScale();
+    this._updateAxis();
+    this._updatePoints();
 };
 
 /**
- * setScale: Set the scale of the plot.
+ * updatePoint: Update scatterplot point.
  *
- * Set the plot scale. Must be called before axis or plot points can be drawn.
+ * When this method is called, an ajax request is sent to get the IDs of the
+ * neighbouring points from the database. Once this request is complete the plot
+ * stylings are updated accordingly.
  *
  * @this {NeighbourPlot}
+ * @param {string} sampleId - The unique _id of the sample that was clicked.
  */
-NeighbourPlot.prototype.setScale = function() {
-    var self = this;
-    // get minimum and maximum values for the x and y axis
-    var xMinMax = d3.extent(this.sampleData, function(d) {
-        return d.dimension_reduce[self.reduction_type][0];
-    });
-    var yMinMax = d3.extent(this.sampleData, function(d) {
-        return d.dimension_reduce[self.reduction_type][1];
+NeighbourPlot.prototype.updatePoint = function(sampleId) {
+    var selectedPoint = this.svg.select(Utils.makeSelector(sampleId));
+
+    $.ajax({
+        url: '/api/samples/?id=' + sampleId + '&select=neighbours',
+        success: function(data) {
+            // create an array of DOM selectors from the list of neighbours
+            var neighbourTags = _.map(data[0].neighbours, Utils.makeSelector);
+
+            this.svg.selectAll('.activept')
```

Comment:
I'm guessing this statement and the next one remove the current selection. I would say that in a comment or even make a new function, e.g. `this.clearActiveSelection`.